lib/src/main/scala/spinal/lib/bus/amba3/ahblite/AhbLite3Arbiter.scala

```diff
@@ -53,42 +72,71 @@ case class AhbLite3Arbiter(ahbLite3Config: AhbLite3Config, inputsCount: Int, rou
     val maskProposal    = Bits(inputsCount bits)
     val maskLocked      = Reg(Bits(inputsCount bits)) init(BigInt(1) << (inputsCount - 1))
     val maskRouted      = Mux(locked || dataPhaseActive, maskLocked, maskProposal)
+    val requestIndex    = OHToUInt(maskRouted)
+
+    /** Backup address phase */
+    val addressPhaseData  = Vec(Reg(AhbLite3AddrPhase(ahbLite3Config)), io.inputs.length)
+    val addressPhaseValid = Vec(RegInit(False), io.inputs.length)
+
+    for((input, index) <- io.inputs.zipWithIndex){
+      when(input.HSEL & input.HTRANS === 2 & input.HREADYOUT){
+        addressPhaseData(index).assignFromBus(io.inputs(index))
+        addressPhaseValid(index) := True
+      }
+    }
```

Comment:
for consistency with previous widths, use `inputsCount`. Zip the three parts together for simplicity.
```scala
val addressPhaseData = Vec(Reg(AhbLite3AddrPhase(ahbLite3Config)), inputsCount)
val addressPhaseValid = Vec(RegInit(False), inputsCount)

for ((input, data, valid) <- (io.inputs, addressPhaseData, addressPhaseValid).zipped) {
  when(input.HSEL & input.HTRANS === 2 & input.HREADYOUT) {
    data.assignFromBus(input)
    valid := True
  }
}
```